client/x/evmstaking/keeper/staking_queue_test.go

```diff
@@ -0,0 +1,138 @@
+func (s *TestSuite) setupValidatorAndDelegation(ctx sdk.Context, valPubKey, delPubKey crypto.PubKey, valAddr sdk.ValAddress, delAddr sdk.AccAddress) {
+	require := s.Require()
+	bankKeeper, stakingKeeper, keeper := s.BankKeeper, s.StakingKeeper, s.EVMStakingKeeper
+
+	// Convert public key to cosmos format
+	valCosmosPubKey, err := k1util.PubKeyToCosmos(valPubKey)
+	require.NoError(err)
+
+	// Create and update validator
+	val := testutil.NewValidator(s.T(), valAddr, valCosmosPubKey)
+	valTokens := stakingKeeper.TokensFromConsensusPower(ctx, 10)
+	validator, _ := val.AddTokensFromDel(valTokens)
+	bankKeeper.EXPECT().SendCoinsFromModuleToModule(gomock.Any(), stypes.NotBondedPoolName, stypes.BondedPoolName, gomock.Any())
+	_ = skeeper.TestingUpdateValidator(stakingKeeper, ctx, validator, true)
+
+	// Create and set delegation
+	delAmt := stakingKeeper.TokensFromConsensusPower(ctx, 100).ToLegacyDec()
+	delegation := stypes.NewDelegation(delAddr.String(), valAddr.String(), delAmt)
+	require.NoError(stakingKeeper.SetDelegation(ctx, delegation))
+
+	// Map delegator to EVM address
+	delEvmAddr, err := k1util.CosmosPubkeyToEVMAddress(delPubKey.Bytes())
+	require.NoError(err)
+	require.NoError(keeper.DelegatorMap.Set(ctx, delAddr.String(), delEvmAddr.String()))
+
+	// Ensure delegation is set correctly
+	delegation, err = stakingKeeper.GetDelegation(ctx, delAddr, valAddr)
+	require.NoError(err)
+	require.Equal(delAmt, delegation.GetShares())
+}
+
+// setupUnbonding creates unbondings for testing.
+func (s *TestSuite) setupUnbonding(ctx context.Context, delAddr sdk.AccAddress, valAddr sdk.ValAddress, amount string) {
+	require := s.Require()
+	bankKeeper, stakingKeeper := s.BankKeeper, s.StakingKeeper
+
+	bankKeeper.EXPECT().SendCoinsFromModuleToModule(gomock.Any(), stypes.BondedPoolName, stypes.NotBondedPoolName, gomock.Any())
+	_, _, err := stakingKeeper.Undelegate(ctx, delAddr, valAddr, math.LegacyMustNewDecFromStr(amount))
+	require.NoError(err)
+}
+
+func (s *TestSuite) TestGetMatureUnbondedDelegations() {
+	require := s.Require()
+	ctx, keeper, stakingKeeper := s.Ctx, s.EVMStakingKeeper, s.StakingKeeper
+
+	// Setup unbonding period
+	params, err := stakingKeeper.GetParams(ctx)
+	require.NoError(err)
+	params.UnbondingTime, err = time.ParseDuration("3600s")
+	require.NoError(err)
+	require.NoError(stakingKeeper.SetParams(ctx, params))
+
+	// Setup 2 validators
+	pubKeys, accAddrs, valAddrs := createAddresses(3)
+	delAddr := accAddrs[0]
+	delPubKey := pubKeys[0]
+	valPubKey := pubKeys[1]
+	valAddr := valAddrs[1]
+	valPubKey2 := pubKeys[2]
+	valAddr2 := valAddrs[2]
+	s.setupValidatorAndDelegation(ctx, valPubKey, delPubKey, valAddr, delAddr)
+	s.setupValidatorAndDelegation(ctx, valPubKey2, delPubKey, valAddr2, delAddr)
+
+	tcs := []struct {
+		name           string
+		setUnbondings  func(c context.Context)
+		expectedResult []stypes.DVPair
+		expectedError  string
+	}{
+		{
```

Comment:
Can we add a test case with mixed items of items within the unbonding period and items after the unbonding period?

Reply:
@limengformal 

<img width="448" alt="image" src="https://github.com/user-attachments/assets/11f82c20-f680-47e8-bd71-6edf9cbddd4e">

In the situation shown above, you’re referring to a test case where GetMaturedUnbondings returns ubd1 and ubd2, right? Since ubd3 and ubd4 have not yet matured, they should not appear in the result of GetMaturedUnbondings; only ubd1 and ubd2 should be shown explicitly

Reply:
Yes, that is the scenarios I was referring to. Let's add that

Reply:
@limengformal 
I have rebased latest main and add [mixed test case](https://github.com/piplabs/story/pull/68/files#diff-60aae4ca5374cd765a3a9b121e416c91c5bbbf2815b2907965feb46c164cd346R79-R88).